packages/website/docs/components/patterns/AI Writing Assistant/main.js

```diff
@@ -0,0 +1,218 @@
+const icon = document.getElementById('ai-wa-icon');
+const menuGenerate = document.getElementById('ai-wa-menu-generate');
+const menuRegenerate = document.getElementById('ai-wa-menu-regenerate');
+
+let inputValues = [];
+let currentInputIndex = -1;
+let loadTimeout;
+
+
+input.addEventListener('focus', () => {
+	updateIconVisibility(true);
+	updateIconState("ai", "AI Writing Assistant (Shift + F4)");
+});
+
+input.addEventListener('focusout', e => {
+	if (!menuGenerate.open && !menuRegenerate.open) {
+		updateIconState("ai", "AI Writing Assistant (Shift + F4)");
+		updateIconVisibility(false);
+	}
+});
+
+icon.addEventListener("mousedown", e => {
+	if (icon.name === "stop") {
+		stopLoading();
+		return;
+	}
+	if (input.value.length) {
+		updateMenuVisibility(menuRegenerate, true, true);
+		updateMenuVisibility(menuGenerate, false);
+
+	} else {
+		updateMenuVisibility(menuGenerate, true, true);
+		updateMenuVisibility(menuRegenerate, false);
+	}
+});
+
+menuGenerate.addEventListener("before-close", e => {
+	if (icon.name === "stop") {
+		return;
+	}
+	updateIconVisibility(false);
+});
+
+menuRegenerate.addEventListener("before-close", e => {
+	if (icon.name === "stop") {
+		return;
+	}
+	updateIconVisibility(false);
+});
+
+
+menuGenerate.addEventListener("item-click", (e) => {
+	onMenuItemClick(e.detail.text);
+});
+
+
+menuRegenerate.addEventListener("item-click", (e) => {
+	onMenuItemClick(e.detail.text);
+});
+
+
+document.addEventListener('keydown', function (event) {
+	const isMac = navigator.userAgent.includes('Mac')
+	if (event.key === 'Escape') {
+		stopLoading();
+	}
+
+	if (event.key === 'F4' && event.shiftKey) {
+		if (input.value.length) {
+			updateMenuVisibility(menuRegenerate, true, true);
+			updateMenuVisibility(menuGenerate, false);
+		} else {
+			updateMenuVisibility(menuGenerate, true, true);
+			updateMenuVisibility(menuRegenerate, false);
+		}
+	}
+
+	if (event.key === 'Enter' && (event.ctrlKey || (isMac && event.metaKey))) {
+		transitionToLoadState();
+	}
+
+	if (event.key === 'Z' && (event.ctrlKey || (isMac && event.metaKey)) && event.shiftKey) {
+		if (inputValues.length > 1 && currentInputIndex > 0) {
+			currentInputIndex--;
+			const previousValue = inputValues[currentInputIndex];
+			updateInputState(true, previousValue, 'Placeholder');
+		} else {
+			console.log("No previous values available.");
+		}
+	}
+	if (event.key === 'Y' && (event.ctrlKey || (isMac && event.metaKey)) && event.shiftKey) {
+		if (inputValues.length > 0 && currentInputIndex < inputValues.length - 1) {
+			currentInputIndex++;
+			const nextValue = inputValues[currentInputIndex];
+			updateInputState(true, nextValue, 'Placeholder');
+		} else {
+			console.log("No next values available.");
+		}
+	}
+});
+
+function onMenuItemClick(text) {
+	switch (text) {
+		case "Generate":
+			transitionToLoadState();
+			break;
+		case "Regenerate":
+			transitionToLoadState();
+			break;
+		case "Fix spelling and grammar":
```

Comment:
I think it's better to use something like: "Correct spelling and grammar" or "Check spelling and grammar"

Reply:
This part was removed